Automattic/wp-calypso
# Odyssey Stats: Fetch purchases for displaying notice correctly

Related to https://github.com/Automattic/jetpack/issues/31712

## Proposed Changes
The subscription `Jetpack Stats (Commercial license)` removal was not updated immediately after the operation.

* Fetch current purchases via API endpoint `jetpack/v4/site/purchases` to promptly display the notice for Paid Stats.

## Testing Instructions



* Build Jetpack if necessary.
  * `jetpack build plugins/jetpack`.
* Launch Odyssey Stats on the local Jetpack site.
  * `STATS_PACKAGE_PATH=/path/to/jetpack/projects/packages/stats-admin yarn dev`
* Navigate to `/wp-admin/admin.php?page=stats&flags=stats/paid-stats`.
* Ensure the `Do you love Jetpack Stats` notice display correctly.
* Operate removing or purchasing the `Jetpack Stats (Commercial license)` product.
  * removing - `https://cloud.jetpack.com/purchases/subscriptions/${jetpack-site-slug}`
  * purchasing - `https://wordpress.com/checkout/${jetpack-site-slug}/jetpack_stats_monthly`
* Ensure the `Do you love Jetpack Stats` notice displays according to the current subscription status immediately.
* Ensure the changes also work as described above on Calypso Stats.

## Pre-merge Checklist
